### PR TITLE
feat(repo-tools): Rework command structure and standardize `openapi generate-client`

### PR DESCRIPTION
**Pull request by sennyeya**
## Hey, I just made a Pull Request!

work towards https://github.com/backstage/backstage/issues/21790 (@taras 😎)

Updates the structure of the `schema openapi` commands to be more clear about what they're targeting, namely breaking them up into `repo` and `package`. This also makes the `generate-client` command interface a bit nicer, removing the need to supply an input spec since that can be inferred and just making the user provide an output package which is resolved from root,
from:
```
yarn backstage-repo-tools schema openapi generate-client --input-spec plugins/catalog-backend/src/schema/openapi.yaml --output-directory packages/catalog-client
```
to: 
```
yarn backstage-repo-tools package schema openapi generate client --output-package packages/catalog-client
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/repo-tools/src/commands/package/schema/openapi/init.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ async function init(directoryPath: string) {
   const openapiPath = join(directoryPath, YAML_SCHEMA_PATH);
   if (!(await fs.pathExists(openapiPath))) {
     throw new Error(
-      `You do not have an OpenAPI YAML file at ${openapiPath}. Please create one and retry this command. If you already have existing test cases for your router, see 'backstage-repo-tools schema openapi test --update'`,
+      `You do not have an OpenAPI YAML file at ${openapiPath}. Please create one and retry this command. If you already have existing test cases for your router, see 'backstage-repo-tools package schema openapi test --update'`,
```
Comment (Rugvip) [inline packages/repo-tools/src/commands/package/schema/openapi/init.ts:34]:
```suggestion
      `You do not have an OpenAPI YAML file at ${openapiPath}. Please create one and retry this command. If you already have existing test cases for your router, see 'backstage-repo-tools repo schema openapi test --update'`,
```

Right?

File: packages/repo-tools/src/commands/index.ts [hunk 2]
```diff
@@ -18,31 +18,83 @@ import { assertError } from '@backstage/errors';
...
 
-  const openApiCommand = command
+  const openApiCommand = schemaCommand
     .command('openapi [command]')
-    .description('Tooling for OpenApi schema');
+    .description('Tooling for OpenAPI schema');
 
   openApiCommand
-    .command('verify [paths...]')
+    .command('init')
```
Comment (Rugvip) [inline packages/repo-tools/src/commands/index.ts:37]: Should have `<paths...>` as well right?

And this isn't really a package command right? Looks like it's meant to be run from the repo root

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 😁

File: packages/repo-tools/src/commands/package/schema/openapi/init.ts (hunk 1 above)
Comment (sennyeya) [reply packages/repo-tools/src/commands/package/schema/openapi/init.ts:34]: Yup 😅

File: packages/repo-tools/src/commands/index.ts (hunk 2 above)
Comment (sennyeya) [reply packages/repo-tools/src/commands/index.ts:37]: My goal is for you to be able to run `init` in a single package to start onboarding to the tooling. I wouldn't _expect_ people to onboard multiple packages at the same time. diff here is a little weird, `verify` should be still repo level.

File: plugins/catalog-backend/package.json [hunk 3]
```diff
@@ -42,7 +42,9 @@
     "test": "backstage-cli package test",
     "prepack": "backstage-cli package prepack",
     "postpack": "backstage-cli package postpack",
-    "clean": "backstage-cli package clean"
+    "clean": "backstage-cli package clean",
+    "generate-server": "backstage-repo-tools package schema openapi generate server",
+    "generate-client": "backstage-repo-tools package schema openapi generate client --output-package packages/catalog-client"
```
[the remark this answers is not in the document]
Comment (sennyeya) [reply plugins/catalog-backend/package.json:47]: yes, I think it makes sense at this point. The npm resolve would be cool, but let's start with this and maybe add that as an additional resolve feature in the future?

File: packages/repo-tools/src/commands/index.ts (hunk 2 above)
Comment (sennyeya) [reply packages/repo-tools/src/commands/index.ts:37]: Updated in https://github.com/backstage/backstage/pull/22350/commits/2a42f97175f2c04e7a4a60ea24150a8156a24ac3

File: plugins/catalog-backend/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sennyeya) [reply plugins/catalog-backend/package.json]: Updated in https://github.com/backstage/backstage/pull/22350/commits/b107fdf6090352c2c1715891b0f86e46fec67d0c

**Review by Rugvip — APPROVED** [review]
Sweet, let's :shipit: ! 🎉

**Pull request merged**